src/test/java/org/fcrepo/upgrade/utils/F47ToF5UpgradeManagerTest.java

```diff
@@ -0,0 +1,113 @@
+        final File tmpDir = new File(System.getProperty("java.io.tmpdir"), System.currentTimeMillis() + "");
+        tmpDir.mkdirs();
+        return tmpDir;
+    }
+
+    @Test
+    public void testUpgrade() throws Exception {
+        //prepare
+        final File tmpDir = createTempDir();
+        final File input = new File(TARGET_DIR + "/test-classes/4.7.5-export");
+        final File output = new File(tmpDir, "output");
+        output.mkdir();
+
+        final var config = new Config();
+        config.setSourceVersion(FedoraVersion.V_4_7_5);
+        config.setTargetVersion(FedoraVersion.V_5);
+        config.setInputDir(input);
+        config.setOutputDir(output);
+        //run
+        UpgradeManager upgradeManager = UpgradeManagerFactory.create(config);
+        upgradeManager.start();
+        //ensure all expected files exist
+        final String[] expectedFiles = new String[]{"rest.ttl",
+                "rest.ttl.headers",
+                "rest/external1",
+                "rest/external1/fcr%3Ametadata.ttl",
+                "rest/container1.ttl.headers",
+                "rest/container1/testbinary.binary",
+                "rest/container1/testbinary/fcr%3Ametadata.ttl",
+                "rest/container1/testbinary.binary.headers",
+                "rest/container1.ttl",
+                "rest/external1.external.headers",
+                "rest/external1.external"};
+
+        for (String f : expectedFiles) {
+            assertTrue(f + " does not exist as expected", getFile(output, f).exists());
+        }
+
+        //ensure external content has been transformed properly
+        try (FileInputStream fis = new FileInputStream(getFile(output, "rest/external1/fcr%3Ametadata.ttl"))) {
+            final String externalContent = IOUtils.toString(fis, "UTF-8");
+            assertFalse("external content metadata should contain the mimetype", externalContent.contains("image/jpg"));
+            assertFalse("message/external-body should not be present in the external content metadata",
+                    externalContent.contains("message/external-body"));
+
+        }
+
+        //ensure the binaries contain NonRDFSource types in their headers
+        final Map<String, List<String>> bheadders =
+                deserializeHeaders(getFile(output, "rest/container1/testbinary.binary.headers"));
+        assertTrue("binary does not contain NonRDFSource type in the link headers",
+                bheadders.get("Link").stream().anyMatch(x -> x.contains("NonRDFSource")));
+
+    }
+
+    private Map<String, List<String>> deserializeHeaders(final File headerFile) throws IOException {
+        final byte[] mapData = Files.readAllBytes(Paths.get(headerFile.toURI()));
+        final ObjectMapper objectMapper = new ObjectMapper();
+        return objectMapper.readValue(mapData, new TypeReference<HashMap<String, List<String>>>() {
+        });
+    }
+
+    private File getFile(final File output, final String subpath) {
```

Comment:
You can probably replace `getFile` with `new File(output, subpath)`, or at least update the body of this method to use that format.